mktrayicon.c

```diff
@@ -19,12 +19,39 @@
  */
 char* strncpy_esc(char *dest, const char *src, size_t n)
 {
-	size_t i;
+	size_t i = 0;
 	size_t index = 0;
-	for (i = 0; i < n && src[i] != '\0'; i++){
-		if (src[i] != '\\'){
+	while (i < n && src[i] != '\0'){
+		if (src[i] == '\''){ 
```

Comment:
I think you don't need this entire special-case. Instead, just remember if the first character was a `'`. Then, copy the string as before (skipping `\`). When you hit the end of the string (`\0`), if the first character was a `'`, check that the last character of `dest` is now a `'`. Then, turn it into a `\0`.

Reply:
The logic is that we can input an `src` that is not all in quotes and it will parse it correctly. So there are cases that the first char is not a quotes but there are quotes inside.
For example, `test'|||'end` will be parsed correctly, as one word `test|||end`

Reply:
I think it's totally fine for us to not support that, and only support whole-sale quoting of the argument. It'll make the code nicer without (I think) removing any particularly important use-cases.

Reply:
I think it's trickier (not much, but still) to remove this "behavior". What I can look into is changing the function and using this nice idea of remembering what we are reading. Much like the `outQuoted` (or `inQuoted`) variable. You are right that it is not so clean and readable now, so it should be written to be more intuitive